### PR TITLE
Avatar envmaps hotfix

### PR DESCRIPTION
**Pull request by AidanCaruso**
### <samp>🤖 Generated by Copilot at e9c5981</samp>

This pull request adds a new feature to the engine that allows avatars to have environment maps applied to them. It also refactors some code to reuse existing functions and improve readability.

## References
closes #_insert number here_

### <samp>🤖 Generated by Copilot at e9c5981</samp>

*  Add a function to set up environment maps for avatar models ([link](https://github.com/EtherealEngine/etherealengine/pull/8908/files?diff=unified&w=0#diff-ae0c696fdcb759ddeda5f76ab4b6722ad7279c8c65438f5008fd6fadfb9535a9L169-R193))
  - Import constants, components, and functions from `three` and `@xrengine/engine` packages ([link](https://github.com/EtherealEngine/etherealengine/pull/8908/files?diff=unified&w=0#diff-ae0c696fdcb759ddeda5f76ab4b6722ad7279c8c65438f5008fd6fadfb9535a9R33), [link](https://github.com/EtherealEngine/etherealengine/pull/8908/files?diff=unified&w=0#diff-ae0c696fdcb759ddeda5f76ab4b6722ad7279c8c65438f5008fd6fadfb9535a9R51), [link](https://github.com/EtherealEngine/etherealengine/pull/8908/files?diff=unified&w=0#diff-ae0c696fdcb759ddeda5f76ab4b6722ad7279c8c65438f5008fd6fadfb9535a9L57-R59), [link](https://github.com/EtherealEngine/etherealengine/pull/8908/files?diff=unified&w=0#diff-ae0c696fdcb759ddeda5f76ab4b6722ad7279c8c65438f5008fd6fadfb9535a9R69-R70))
  - Query for the entity with the `EnvMapBakeComponent` and load the texture from the URL ([link](https://github.com/EtherealEngine/etherealengine/pull/8908/files?diff=unified&w=0#diff-ae0c696fdcb759ddeda5f76ab4b6722ad7279c8c65438f5008fd6fadfb9535a9L169-R193))
  - Set the mapping type of the texture to `EquirectangularReflectionMapping` ([link](https://github.com/EtherealEngine/etherealengine/pull/8908/files?diff=unified&w=0#diff-ae0c696fdcb759ddeda5f76ab4b6722ad7279c8c65438f5008fd6fadfb9535a9L169-R193))
  - Call the `updateEnvMap` function with the avatar entity's `GroupComponent` and the texture ([link](https://github.com/EtherealEngine/etherealengine/pull/8908/files?diff=unified&w=0#diff-ae0c696fdcb759ddeda5f76ab4b6722ad7279c8c65438f5008fd6fadfb9535a9L169-R193))
  - Call the `setupAvatarEnvmaps` function from the `setupAvatarForUser` function ([link](https://github.com/EtherealEngine/etherealengine/pull/8908/files?diff=unified&w=0#diff-ae0c696fdcb759ddeda5f76ab4b6722ad7279c8c65438f5008fd6fadfb9535a9L169-R193))
* Export the `updateEnvMap` function from the `EnvmapComponent.tsx` file ([link](https://github.com/EtherealEngine/etherealengine/pull/8908/files?diff=unified&w=0#diff-c3b5fd77b5cfb8085e215c2bafd90a2727de2d03ad82fe74db17b47605762c7eL197-R197))
  - Allow the function to be imported and used in the `avatarFunctions.ts` file ([link](https://github.com/EtherealEngine/etherealengine/pull/8908/files?diff=unified&w=0#diff-ae0c696fdcb759ddeda5f76ab4b6722ad7279c8c65438f5008fd6fadfb9535a9R69-R70))
* Remove an empty line of code from the `avatarFunctions.ts` file ([link](https://github.com/EtherealEngine/etherealengine/pull/8908/files?diff=unified&w=0#diff-ae0c696fdcb759ddeda5f76ab4b6722ad7279c8c65438f5008fd6fadfb9535a9R164))

### <samp>🤖 Generated by Copilot at e9c5981</samp>

> _Sing, O Muse, of the skillful coder who devised_
> _A wondrous map of the environs for the avatar,_
> _The shining image of the self, and made it shine_
> _With textures fetched from distant URLs, the work of `updateEnvMap`._

## QA Steps
_List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos._

## Checklist
- If this PR is still a WIP, convert to a draft
- When this PR is ready, mark it as "Ready for review"
- [ensure all checks pass](https://github.com/etherealengine/etherealengine/wiki/Testing-&-Contributing)
- Changes have been manually QA'd 
- Changes reviewed by at least 2 approved reviewers

### CODE REVIEW

File: packages/engine/src/avatar/functions/avatarFunctions.ts [hunk 1]
```diff
@@ -166,9 +171,26 @@ export const setupAvatarForUser = (entity: Entity, model: VRM) => {
   createIKAnimator(entity)
 
   setObjectLayers(model.scene, ObjectLayers.Avatar)
+  setupAvatarEnvmaps(entity, model.scene)
   avatar.model = model.scene
 }
 
+const setupAvatarEnvmaps = (entity: Entity, model: Object3D) => {
+  //query all envmap bake components and get the url from the first one
+  const envmapBakes = defineQuery([EnvMapBakeComponent])
```
Comment (HexaField) [inline packages/engine/src/avatar/functions/avatarFunctions.ts:180]: query should be in module scope

File: packages/engine/src/avatar/functions/avatarFunctions.ts [hunk 2]
```diff
@@ -166,9 +171,26 @@ export const setupAvatarForUser = (entity: Entity, model: VRM) => {
...
+  const envmapBake = getComponent(envmapBakes()[0], EnvMapBakeComponent)
+  console.log(envmapBake)
```
Comment (HexaField) [inline packages/engine/src/avatar/functions/avatarFunctions.ts:182]: cleanup console logs

**Review by HexaField — CHANGES_REQUESTED** [review]
curious why we need to call updateEnvMap manually for avatars, as opposed to just adding the component and letting it's reactor handle things

**Comment by AidanCaruso** [comment]
> curious why we need to call updateEnvMap manually for avatars, as opposed to just adding the component and letting it's reactor handle things

Tried doing this originally because it's the no brainer solution. Could be a race condition, or maybe object3d/group weirdness. Didn't have time tonight to debug that, just wanted lighting/reflections to look pretty. There's a separate issue with the spawn effect shader that breaks envmaps but it's not a problem right now because it seems it's hard coded off.

**Comment by HexaField** [comment]
> > curious why we need to call updateEnvMap manually for avatars, as opposed to just adding the component and letting it's reactor handle things
> 
> Tried doing this originally because it's the no brainer solution. Could be a race condition, or maybe object3d/group weirdness. Didn't have time tonight to debug that, just wanted lighting/reflections to look pretty. avatarFunctions seriously needs a refactor anyway. There's also a separate issue with the spawn effect shader that breaks envmaps but it's not a problem right now because it seems it's hard coded off.

my guess is we need to just listen for changes to group.length instead of group in the reactor, ill give it a try

**Comment by HexaField** [comment]
migrated changes to https://github.com/EtherealEngine/etherealengine/pull/8734

**Pull request closed without merging**